### PR TITLE
SNOW-1833578 Add net9 docker image

### PR DESCRIPTION
**Pull request by sfc-gh-knozderko**
### Description
SNOW-1833578 Add net9 docker image

### Checklist
- [x] Code compiles correctly
- [x] Code is formatted according to [Coding Conventions](../blob/master/CodingConventions.md)
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing (`dotnet test`)
- [x] Extended the README / documentation, if necessary
- [x] Provide JIRA issue id (if possible) or GitHub issue id in PR name

### CODE REVIEW

**Review by sfc-gh-pmotacki — APPROVED** [review]

File: ci/image/Dockerfile.dotnet-ubuntu204-net9-build [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+#  install basic tools
+RUN apt-get install -y git
+RUN apt-get install -y zstd
+RUN apt-get install -y jq
+
+# Install .NET SDK 9.0
+RUN apt-get update
+RUN apt-get install -y
+RUN apt-get install -y dotnet-sdk-6.0
```
Comment (sfc-gh-akolodziejczyk) [inline ci/image/Dockerfile.dotnet-ubuntu204-net9-build:37]: do we need to install dotnet 6,7 and 9 sdk?

File: ci/_init.sh [hunk 2]
```diff
@@ -21,11 +21,11 @@ BUILD_IMAGE_VERSION=1
 TEST_IMAGE_VERSION=1
 
 declare -A BUILD_IMAGE_NAMES=(
-    [$DRIVER_NAME-ubuntu204-net8]=$DOCKER_REGISTRY_NAME/client-$DRIVER_NAME-ubuntu204-net8-build:$BUILD_IMAGE_VERSION
```
Comment (sfc-gh-akolodziejczyk) [inline ci/_init.sh:24]: Will the Net 8 image still be used? maybe we can keep both in the table ?

File: ci/image/Dockerfile.dotnet-ubuntu204-net9-build (hunk 1 above)
Comment (sfc-gh-knozderko) [reply ci/image/Dockerfile.dotnet-ubuntu204-net9-build:37]: I think I could not install them but installing them does not do anything bad especially that in net8 image there was also net6 sdk installed.

Comment (sfc-gh-akolodziejczyk) [reply ci/image/Dockerfile.dotnet-ubuntu204-net9-build:37]: If it's not needed, I would just not install them and have a smaller image

File: ci/_init.sh (hunk 2 above)
Comment (sfc-gh-akolodziejczyk) [reply ci/_init.sh:24]: ok, in case we need to modify net8 we can just add it

File: ci/image/Dockerfile.dotnet-ubuntu204-net9-build (hunk 1 above)
Comment (sfc-gh-knozderko) [reply ci/image/Dockerfile.dotnet-ubuntu204-net9-build]: I removed net 6.0 but preserved net 8.0

**Review by sfc-gh-akolodziejczyk — APPROVED** [review]

File: ci/image/Dockerfile.dotnet-ubuntu204-net9-build (hunk 1 above)
Comment (sfc-gh-knozderko) [reply ci/image/Dockerfile.dotnet-ubuntu204-net9-build]: tests are failing without net 6.0, so I restored it

**Pull request merged**